cobra/test/test_model.py

```diff
@@ -267,6 +295,51 @@ def benchmark_add_reaction():
 
         benchmark(benchmark_add_reaction)
 
+    def test_add_metabolite(self, model):
+        new_metabolite = Metabolite('test_met')
+        assert new_metabolite not in model.metabolites
+        with model:
+            model.add_metabolites(new_metabolite)
+            assert new_metabolite._model == model
+            assert new_metabolite in model.metabolites
+            assert new_metabolite.id in model.solver.constraints
+
+        assert new_metabolite._model is None
+        assert new_metabolite not in model.metabolites
+        assert new_metabolite.id not in model.solver.constraints
+
+    def test_remove_metabolite_subtractive(self, model):
+        test_metabolite = model.metabolites[4]
+        test_reactions = test_metabolite.reactions
+        with model:
+            model.remove_metabolites(test_metabolite, method='subtractive')
+            assert test_metabolite._model is None
+            assert test_metabolite not in model.metabolites
+            assert test_metabolite.id not in model.solver.constraints
+            for reaction in test_reactions:
+                assert reaction in model.reactions
+
+        assert test_metabolite._model is model
+        assert test_metabolite in model.metabolites
+        assert test_metabolite.id in model.solver.constraints
+
+    def test_remove_metabolite_destructive(self, model):
+        test_metabolite = model.metabolites[4]
+        test_reactions = test_metabolite.reactions
+        with model:
+            model.remove_metabolites(test_metabolite, method='destructive')
```

Comment:
Good point. Kept it for compatibility, but it was never in the tests. In fact, `metabolite.remove_from_model(method='destructive')` was broken until this fix (at least on python 3), so I doubt anyone was using it.